src/main/java/bio/terra/pdao/bigquery/BigQueryPdao.java

```diff
@@ -0,0 +1,138 @@
+package bio.terra.pdao.bigquery;
+
+import bio.terra.metadata.Study;
+import bio.terra.metadata.StudyTable;
+import bio.terra.metadata.StudyTableColumn;
+import bio.terra.pdao.PdaoException;
+import bio.terra.pdao.PrimaryDataAccess;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.DatasetId;
+import com.google.cloud.bigquery.DatasetInfo;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardTableDefinition;
+import com.google.cloud.bigquery.TableDefinition;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableInfo;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+@Profile("bigquery")
+public class BigQueryPdao implements PrimaryDataAccess {
+
+    private final BigQuery bigQuery;
+    private final String projectId;
+    private final String rowIdColumnName;
+    private final String rowIdColumnDatatype;
+
+    public BigQueryPdao(
+            @Autowired BigQuery bigQuery,
```

Comment:
@ddietterich what is the advantage to autowiring at the param level instead of `@Autowire`ing the constructor?

Reply:
I don't think there is one. It should be pulled out onto the class.
Cut'n'paste laziness...